ols/src/cache/postgres_cache.py

```diff
@@ -35,8 +127,85 @@ def insert_or_append(
             user_id: User identification.
             conversation_id: Conversation ID unique for given user.
             value: The value to set.
-
-        Raises:
-            OutOfMemoryError: If item is evicted when Redis allocated
-                memory is higher than maxmemory.
         """
+        # the whole operation is run in one transaction
+        with self.conn.cursor() as cursor:
+            old_value = self._select(cursor, user_id, conversation_id)
+            if old_value:
+                old_value.extend(value)
+                PostgresCache._update(
+                    cursor,
+                    user_id,
+                    conversation_id,
+                    pickle.dumps(old_value, protocol=pickle.HIGHEST_PROTOCOL),
+                )
+            else:
+                PostgresCache._insert(
+                    cursor,
+                    user_id,
+                    conversation_id,
+                    pickle.dumps(value, protocol=pickle.HIGHEST_PROTOCOL),
+                )
+                PostgresCache._cleanup(cursor, self.capacity)
```

Comment:
same comment here about error/exception handling.

i'd expect postgres_cache to abstract away the postgres specific errors/exceptions, other than including them in the error message, so that a client can expect to handle the same kinds of errors/exceptions regardless of which cache implementation they have chosen to use.

Reply:
updated